bytecomp/debuginfo.ml

```diff
@@ -54,3 +54,17 @@ let from_location kind loc =
 
 let from_call ev = from_location Dinfo_call ev.Lambda.lev_loc
 let from_raise ev = from_location Dinfo_raise ev.Lambda.lev_loc
+
+let to_location d =
+  if is_none d then Location.none
+  else
+    let loc_start =
+      { Lexing.
+        pos_fname = d.dinfo_file;
+        pos_lnum = d.dinfo_line;
+        pos_bol = 0;
+        pos_cnum = d.dinfo_char_start;
```

Comment:
The whole `bol`/`cnum` dance was to keep compatibility while switching from `cnum`-based code to line/char-based code, especially while the debugger was still using `cnum`. I'm OK with this little bit of weirdness until we get around to refactoring locations (if ever).